content/03-types/02-relational/07-what-are-joins.mdx

```diff
@@ -0,0 +1,401 @@
+| 3           | Synniva | 25  | synniva@prisma.io | NULL     | NULL    | NULL       |
+
+In this case, the result would be the same as a `LEFT JOIN` as every record from the `customers` table has a match in the `orders` table.
+
+When depicted as a Venn diagram, a `FULL JOIN` represents both circles. In other words, a `FULL JOIN` combines all records from both tables.
+
+![full-join](../../dataguide-images/types/relational/what-are-joins/full-join.png)
+
+#### **CROSS JOIN**
+
+A `CROSS JOIN` returns the Cartesian product of the two tables, meaning every row from the first table is combined with every row from the second table. In this syntax, the result is formed by adding each of the rows in the first table with each of the rows from the second table like so:
+
+```sql
+t1.r1 + t2.r1
+t1.r1 + t2.r2
+t1.r1 + t2.r3
+t1.r2 + t2.r1
+t1.r2 + t2.r2
+t1.r2 + t2.r3
+t1.r3 + t2.r1
+t1.r3 + t2.r2
+t1.r3 + t2.r3
+```
+
+> **Note:** In MySQL, the concept of a `CROSS JOIN` is combined with the `INNER JOIN`. Read more in the [data guide](https://www.prisma.io/dataguide/mysql/reading-and-querying-data/joining-tables#different-types-of-joins).
+
+
+With a `CROSS JOIN`, each row from the `customers` table is combined with each row from the `orders` table, resulting in 9 rows. However, we won’t show the result table as a `CROSS JOIN` wouldn’t accurately pair the customer with their respective orders.
+
+
+To learn more about JOINs in PostgreSQL, read more in the data guide [here](https://www.prisma.io/dataguide/postgresql/reading-and-querying-data/joining-tables). To learn more about JOINs in MySQL, read more in the data guide [here](https://www.prisma.io/dataguide/mysql/reading-and-querying-data/joining-tables).
+
+
+### LATERAL JOINs
+
+Lateral JOINs offer a different syntax to combining tables compared to traditional JOINs. In a `LATERAL JOIN`, the second table is presented as a subquery, and the JOIN criteria is defined within the WHERE clause of the subquery.
+
+According to the [PostgreSQL](https://www.postgresql.org/docs/9.3/sql-select.html#SQL-FROM) docs, “The `LATERAL` key word can precede a sub-`SELECT FROM` item. This allows the sub-`SELECT` to refer to columns of `FROM` items that appear before it in the `FROM` list. (Without `LATERAL`, each sub-`SELECT` is evaluated independently and so cannot cross-reference any other `FROM` item.)”
+
+> **Note:** `LATERAL JOIN`s are supported by Postgres but only supported in [MySQL versions >8](https://dev.mysql.com/blog-archive/support-for-lateral-derived-tables-added-to-mysql-8-0-14/)
+
+In other words, a `LATERAL JOIN` is like a foreach loop, where PostgreSQL iterates over each row in a result set and uses each row to evaluate the subquery.
+
+Here is the associated query for the `customers` and `orders` table:
+
+```sql
+SELECT
+    *
+FROM
+    customers
+	LEFT JOIN LATERAL
+	    (SELECT *
+       FROM orders
+       WHERE orders.customer_id = customers.customer_id
+       ) AS alias ON TRUE;
+
+```
```

Comment:
proper formatting according to https://sqlformat.org:

```suggestion
```sql
SELECT *
FROM customers
LEFT JOIN LATERAL
  (SELECT *
   FROM orders
   WHERE orders.customer_id = customers.customer_id ) AS ALIAS ON TRUE;
```
```